evalml/tests/component_tests/test_target_encoder.py

```diff
@@ -0,0 +1,147 @@
+import numpy as np
```

Comment:
I think we should add some unit tests with custom indices. I am now officially paranoid 😂